OpenAI-DotNet/ThreadMessages/ThreadMessageFile.cs

```diff
@@ -0,0 +1,33 @@
+using System.Text.Json.Serialization;
+
+namespace OpenAI.ThreadMessages
+{
+    public class ThreadMessageFile
+    {
+        /// <summary>
+        /// The identifier, which can be referenced in API endpoints.
+        /// </summary>
+        /// <returns></returns>
+        [JsonPropertyName("id")]
+        public string Id { get; set; }
+
+        /// <summary>
+        /// The object type, which is always thread.message.file.
+        /// </summary>
+        /// <returns></returns>
+        [JsonPropertyName("object")]
+        public string Object { get; set; } = "thread.message.file";
```

Comment:
This would be set by default from the API response data.

Remove all the public setters for all properties.